pkg/ingester/ingester.go

```diff
@@ -986,6 +1003,15 @@ func (i *Ingester) startPushRequest(ctx context.Context, reqSize int64) (_ conte
 		// If state is already in context, this means we already passed through StartPushRequest for this request.
 		return ctx, false, nil
 	}
+
+	// We try to acquire a permit from the circuit breaker.
+	// If it is not possible, it is because the circuit breaker is open, and a circuitBreakerOpenError is returned.
+	// If it is possible, a permit has to be released by recording either a success or a failure with the circuit
+	// breaker. This is done in finishPushRequest().
+	if err := i.circuitBreaker.tryAcquirePermit(); err != nil {
```

Comment:
What happens if we "acquire permit" from circuit breaker that's not yet active, but when we call `finishPushRequest`, it is already active?

I think we should remember whether we have truly acquired permit here (CB was active), or not, and only report result to CB if we did get permit.

Reply:
This information is now stored in the `pushRequestState` object added to the context.